src/content/docs/bff/samples/bff-blazor.mdx

```diff
@@ -40,58 +42,111 @@ dotnet add package Duende.BFF.Blazor --version 3.0.0
+            // (duendesoftware.com vs localhost), we need to configure the SameSite property to Lax.
+            // Setting it to Strict would cause the authentication cookie not to be sent after loggin in.
+            // The user would have to refresh the page to get the cookie.
+            // Recommendation: Set it to 'strict' if your IDP is on the same site as your BFF.
+            options.Cookie.SameSite = SameSiteMode.Lax;
+        })
+        .AddServerSideSessions() // Add in-memory implementation of server side sessions
+        .AddBlazorServer();
+
+    // Make sure authentication state is available to all components.
+    builder.Services.AddCascadingAuthenticationState();
+
+    builder.Services.AddAuthorization();
+    ```
+
+  </TabItem>
+  <TabItem label="Duende BFF v3">
+  ```csharp
+  // BFF setup for blazor
+  builder.Services.AddBff()
+      .AddServerSideSessions() // Add in-memory implementation of server side sessions
+      .AddBlazorServer();
 
 // Configure the authentication
-builder.Services.AddAuthentication(options =>
-    {
-        options.DefaultScheme = "cookie";
-        options.DefaultChallengeScheme = "oidc";
-        options.DefaultSignOutScheme = "oidc";
-    })
-    .AddCookie("cookie", options =>
-    {
-        options.Cookie.Name = "__Host-blazor";
-
-        // Because we use an identity server that's configured on a different site
-        // (duendesoftware.com vs localhost), we need to configure the SameSite property to Lax. 
-        // Setting it to Strict would cause the authentication cookie not to be sent after loggin in.
-        // The user would have to refresh the page to get the cookie.
-        // Recommendation: Set it to 'strict' if your IDP is on the same site as your BFF.
-        options.Cookie.SameSite = SameSiteMode.Lax;
-    })
-    .AddOpenIdConnect("oidc", options =>
-    {
-        options.Authority = "https://demo.duendesoftware.com";
-        options.ClientId = "interactive.confidential";
-        options.ClientSecret = "secret";
-        options.ResponseType = "code";
-        options.ResponseMode = "query";
-
-        options.GetClaimsFromUserInfoEndpoint = true;
-        options.SaveTokens = true;
-        options.MapInboundClaims = false;
-
-        options.Scope.Clear();
-        options.Scope.Add("openid");
-        options.Scope.Add("profile");
-        options.Scope.Add("api");
-        options.Scope.Add("offline_access");
-
-        options.TokenValidationParameters.NameClaimType = "name";
-        options.TokenValidationParameters.RoleClaimType = "role";
-    });
-
-// Make sure authentication state is available to all components. 
+builder.Services
+.AddAuthentication(options =>
+{
+options.DefaultScheme = "cookie";
+options.DefaultChallengeScheme = "oidc";
+options.DefaultSignOutScheme = "oidc";
+})
+.AddCookie("cookie", options =>
+{
+options.Cookie.Name = "\_\_Host-blazor";
```

Comment:
This tab's formatting is kind of jank.

Reply:
FFS yeah looks like Prettier or WebStorm messed this up. Fixing...